Source/EjectMenu.spoon/init.lua

```diff
@@ -0,0 +1,170 @@
+--- === Eject Menu ===
+---
+--- An much-needed eject menu for your Mac menu bar.
+--- Configurable properties.
+---  ejectAllHotkey
+---
+--- Download: [https://github.com/Hammerspoon/Spoons/raw/master/Spoons/EjectMenu.spoon.zip](https://github.com/Hammerspoon/Spoons/raw/master/Spoons/EjectMenu.spoon.zip)
+
+local obj = {}
+obj.__index = obj
+
+-- Metadata
+obj.name = 'EjectMenu'
+obj.version = '0.1'
+obj.author = 'Mark Juers <mpjuers@gmail.com>'
+obj.homepage = 'https://github.com/Hammerspoon/Spoons'
+obj.license = 'MIT - https://opensource.org/licenses/MIT'
+
+-- EjectMenu:init()
+-- method
+-- Initializes eject menu with available drives.
+--
+-- Returns
+-- nil
+
+function obj:init ()
+    ejectMenu = hs.menubar.new():setTitle("⏏")
+    flagsEvent = hs.eventtap.new(
+        {hs.eventtap.event.types.flagsChanged},
+        function (event) self:changeEjectMenuIcon(event:getFlags()) end
+    )
+    ejectMenu:setMenu(function (mods) return self:initEjectMenu(mods) end)
+end
+
+-- EjectMenu:bindHotKeys(map)
+-- method
+-- Binds hotkeys.
+--
+-- parameters
+-- * map: a table specifying mappings in the format {name = {{{modifiers}, key}}
+--
+-- returns
+-- nil
+--
+function obj:bindHotKeys (map)
+    local def = {
+        ejectAll = function () self:ejectAll() end
+    }
+    hs.spoons.bindHotkeysToSpec(def, map)
+end
+
+-- EjectMenu:changeEjectMenuIcon(mods)
+-- method
+-- Changes eject menu icon depending on which modifiers are held.
+--
+--parameteres
+--* mods: A table containing for which the keys are the modifiers being held
+--  and the values are 'true'.
+--
+-- returns
+-- nil
+
+function obj:changeEjectMenuIcon (mods)
+    if mods:containExactly({'cmd'}) then
+        ejectMenu:setTitle('⮑')
+    elseif mods:containExactly({}) then
+        ejectMenu:setTitle('⏏')
+    end
+    return(0)
+end
+
+-- EjectMenu:ejectAll()
+-- Ejects all external drives.
+--
+-- returns
+-- nil
+
+function obj:ejectAll ()
+    hs.osascript.applescript(
+        'tell application "Finder" to eject (every disk whose ejectable is true and local volume is true and free space is not equal to 0)'
```

Comment:
Is AppleScript better than hs.fs.volume.eject? This spoon also does ejectAll() but with hs.fs.volume : https://github.com/Hammerspoon/Spoons/pull/177/files

Reply:
I don't know that it's any better. AppleScript has given me the best results in the past for ejecting drives consistently and without intervention, though I haven't tried the `hs.fs.volume.eject`.